### PR TITLE
mrc373 User CLI Restructure

### PR DESCRIPTION
**Pull request by EmmaLRussell**
This branch will just contain the restructure of the project to accommodate app and userCLI having sibling gradle configurations, building and pushing the cli image and a dummy cli app which doesn't make any changes. 

Also still need to revert pushing master branch as latest.

### CODE REVIEW

File: src/app/build.gradle [hunk 1]
```diff
@@ -62,13 +45,19 @@ dependencies {
 	testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
 }
 
+test {
+	useJUnitPlatform()
+}
+
+jar{
```
Comment (EmmaLRussell) [inline src/app/build.gradle]: I have no idea why I needed to add this, but it was always being skipped, which meant I couldn't access app classes from userCLI...

File: src/app/src/main/kotlin/org/imperial/mrc/hint/HintApplication.kt [hunk 2]
```diff
@@ -4,7 +4,7 @@ import org.springframework.boot.SpringApplication
 import org.springframework.boot.autoconfigure.SpringBootApplication
 
 @SpringBootApplication
-class HintApplication
+open class HintApplication
```
Comment (EmmaLRussell) [inline src/app/src/main/kotlin/org/imperial/mrc/hint/HintApplication.kt]: Springboot started complaining that Configuration and Bean classes and methods have to be non-final. According to the documentation that is correct, but I'm not sure why it wasn't a problem before?

Comment (hillalex) [reply src/app/src/main/kotlin/org/imperial/mrc/hint/HintApplication.kt]: How bizarre that it wasn't complaining before? Did we have a plugin that wasn't being activated somehow?

File: .travis.yml [hunk 3]
```diff
@@ -5,35 +5,37 @@ matrix:
       - 10
       cache: npm
       install:
-      - npm install codecov -g
-      - npm install --prefix=app/static
-      - npm run build --prefix=app/static
-      script: npm test --prefix=app/static
+      - npm install codecov -g --prefix=src
```
Comment (hillalex) [inline .travis.yml]: I don't think this prefix is needed here?

File: README.md [hunk 4]
```diff
@@ -10,7 +10,8 @@ Requirements:
 1. Run `npm install` from `app/static`
 1. Run `./scripts/run-dependencies.sh` to start a model API
 1. Run `npm run build` from `app/static` to compile front-end dependencies.
-1. Run app from your IDE or by `cd app && ./gradlew :bootRun` to serve the app on port 8080
+
+1. Run app from your IDE or by `cd app && ./gradlew :app:bootRun` to serve the app on port 8080
```
Comment (hillalex) [inline README.md]: I guess should now be `./src/gradlew  -p src :app:bootRun` or `cd src && ./gradlew :app:bootRun`?

File: .travis.yml [hunk 5]
```diff
@@ -5,35 +5,37 @@ matrix:
...
       - ./scripts/run-dependencies.sh
       script:
-      - ./app/gradlew -p app :test
+      - ./src/gradlew -p src app::test
       after_success:
-      - ./app/gradlew -p app :jacocoTestReport
-      - codecov -f app/coverage/test/*.xml
+      - ./src/gradlew -p src app::jacocoTestReport
+      - echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin
+      - (cd src/userCLI; ./scripts/build; ./scripts/push)
```
Comment (hillalex) [inline .travis.yml]: Doesn't really matter but don't need the `cd` any more

**Pull request merged**